hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestReconstructableException.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ipc;
+
+import java.io.IOException;
+
+/**
+ * Test class of ReconstructableException.
+ */
+public class TestReconstructableException extends IOException
+    implements ReconstructableException<TestReconstructableException> {
+
+  private String field1;
+  private String field2;
```

Comment:
make one final, to see what happens

Reply:
I tried to make it final, but the seems not working, since a single string is needed to construct IOException, then the final field can not be initialized.